Top10Words/BookCalculationsLinq.cs

```diff
@@ -1,41 +1,38 @@
-using System.Linq;
-using BenchmarkDotNet.Attributes;
-using BenchmarkDotNet.Engines;
-using BenchmarkDotNet.Jobs;
-
 namespace Top10Words;
 
-public class BookCalculationsLinq
+public class BookCalculationsLinq:IBookCalculations
 {
-    private DataFromBook _dataFromBook;
+    private readonly DataFromBook _dataFromBook;
     private IBookProcessor _bookProcessor;
-    private readonly Consumer _consumer = new Consumer();
     public BookCalculationsLinq()
     {
-        _dataFromBook = new DataFromBook(@"/Users/mariialeos/RiderProjects/Top10Words/Top10Words/Book.txt");
+        var path = "../../../Book.txt";
+        string fullPath = Path.GetFullPath(path);
+        _dataFromBook = new DataFromBook(fullPath);
         _bookProcessor = new BookProcessor();
     }
 
-    // public BookCalculationsLinq(DataFromBook dataFromBook, IBookProcessor bookProcessor)
-    // {
-    //     _dataFromBook = dataFromBook;
-    //     _bookProcessor = bookProcessor;
-    // }
-    [Benchmark]
+    public BookCalculationsLinq(DataFromBook dataFromBook, IBookProcessor bookProcessor)
+    {
+        _dataFromBook = dataFromBook;
+        _bookProcessor = bookProcessor;
+    }
+
     public int GetAllWordsCount()
```

Comment:
Did I understand correctly that it should be swapped with in main method:
var countWords = bookProcessor.GetWordsList(dataFromBook.BookText).Count();

Reply:
No.
I don't see where do you use it. Event it is used somewhere - it is wrong, because it is not part of your interface.
You should use interfaces if possible. If it is not possible - ask yourself: why I can't use interface here? what the problem?
We can discuss it if you need.